compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -1232,6 +1234,7 @@ private ParserRuleContext getNextRule(ParserRuleContext currentCtx, int nextLook
             case CONSTANT_DECL:
             case NIL_TYPE_DESCRIPTOR:
             case COMPOUND_ASSIGNMENT_STMT:
+            case OPTIONAL_TYPE_DESCRIPTOR:
```

Comment:
Since we push to the context stack when entering the optional-type-desc, we need to pop it too, when exiting from it.
We usually do that in the `getNextRule()`.

Reply:
@SupunS I have corrected it please review again.